rgbd/src/server.cpp

```diff
@@ -1,147 +1,34 @@
-        return;
-    }
-
-    rgbd::Image image(img_ptr->image,depth_img_ptr->image,cam_model_,rgb_image_msg->header.frame_id,rgb_image_msg->header.stamp.toSec());
-
-    rgbd_server.send(image);
+Server::Server()
+{
 }
 
 // ----------------------------------------------------------------------------------------
 
-void camInfoCallback(const sensor_msgs::CameraInfoConstPtr& cam_info_msg) {
-    if (!cam_model_.initialized()) {
-        cam_model_.fromCameraInfo(cam_info_msg);
-    }
+Server::~Server()
+{
 }
 
 // ----------------------------------------------------------------------------------------
 
-int main(int argc, char **argv) {
-    ros::init(argc, argv, "rgbd_transport_server");
-
-    ros::NodeHandle nh;
-    ros::NodeHandle nh_private("~");
-
-    double max_fps = 30;
-    nh_private.getParam("max_fps", max_fps);
-
-    // ----- READ RGB STORAGE TYPE
-
-    rgbd::RGBStorageType rgb_type;
-
-    std::string rgb_type_str = "lossless";
-    nh_private.getParam("rgb_storage", rgb_type_str);
-    if (rgb_type_str == "none")
-        rgb_type = rgbd::RGB_STORAGE_NONE;
-    else if (rgb_type_str == "lossless")
-        rgb_type = rgbd::RGB_STORAGE_LOSSLESS;
-    else if (rgb_type_str == "jpg")
-        rgb_type = rgbd::RGB_STORAGE_JPG;
-    else
-    {
-        ROS_ERROR("Unknown 'rgb_storage' type: should be 'none', 'lossless', or 'jpg'.");
-        return 1;
-    }
-
-    // ----- READ DEPTH STORAGE TYPE
-
-    rgbd::DepthStorageType depth_type;
-
-    std::string depth_type_str = "lossless";
-    nh_private.getParam("depth_storage", depth_type_str);
-    if (depth_type_str == "none")
-        depth_type = rgbd::DEPTH_STORAGE_NONE;
-    else if (depth_type_str == "lossless")
-        depth_type = rgbd::DEPTH_STORAGE_LOSSLESS;
-    else if (depth_type_str == "png")
-        depth_type = rgbd::DEPTH_STORAGE_PNG;
-    else
-    {
-        ROS_ERROR("Unknown 'depth_storage' type: should be 'none', 'lossless', or 'png'.");
-        return 1;
-    }
-
-    // ------------------------------
-
-    rgbd_server.initialize(ros::names::resolve("rgbd"), rgb_type, depth_type);
-
-    ros::Subscriber sub_cam_info = nh.subscribe("cam_info", 1, &camInfoCallback);
+void Server::initialize(const std::string& name, RGBStorageType rgb_type, DepthStorageType depth_type)
+{
+    server_rgbd_.initialize(name, rgb_type, depth_type);
+    server_shm_.initialize(name);
+}
 
-    message_filters::Subscriber<sensor_msgs::Image> sub_rgb_sync_(nh, "rgb_image", 1);
-    message_filters::Subscriber<sensor_msgs::Image> sub_depth_sync_(nh, "depth_image", 1);
+// ----------------------------------------------------------------------------------------
 
-    message_filters::Synchronizer<KinectApproxPolicy> sync_(KinectApproxPolicy(10), sub_rgb_sync_, sub_depth_sync_);
-    sync_.registerCallback(boost::bind(&imageCallback, _1, _2));
+void Server::send(const Image& image, bool threaded)
+{
+    server_rgbd_.send(image, threaded);
+    server_shm_.send(image);
```

Comment:
Do we always need to send both?

Reply:
That is the implementation of the server, otherwise you can use one of the child classes. But maybe the behavior of the server class can change when implementing the reconnection.